ts/model/report/ReportScope.ts

```diff
@@ -0,0 +1,13 @@
+export class ReportScope {
+    public view: string;
+    public parameters: {};
+    public rowLimit: number;
+    public withViewDefinitions: boolean;
+
+    constructor(view, parameters = {}, rowLimit = 20, withViewDefinitions = false) {
```

Comment:
If you add `public` in front of these variables you don't need to duplicate their definition in lines 2-5, nor will you need to assign them in lines 8-11. Take a look at https://www.typescriptlang.org/docs/handbook/classes.html#static-properties.

This should reduce the class down to the following:

```typescript
export class ReportScope {
    constructor(public view: any, public parameters = {}, public rowLimit = 20, public withViewDefinitions = false) { }
}
```

Reply:
brilliant, thanks for that